src/jllvm/materialization/CodeGeneratorUtils.hpp

```diff
@@ -66,20 +92,10 @@ class OperandStack
         m_builder.CreateStore(value, alloc);
     }
 
-    StackState saveState() const
-    {
-        return {m_types, m_topOfStack};
-    }
-
-    void restoreState(StackState state)
-    {
-        m_types = std::move(state.m_types);
-        m_topOfStack = state.m_topOfStack;
-    }
-
-    State getHandlerState() const
+    void restoreState(const State& state)
```

Comment:
I think the name `restoreStack` is not suitable anymore now that the `saveState` counterpart is gone.
Maybe just `setState`?